### PR TITLE
HPCC-15204 Ensure execution on another cluster modifies the distribution

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@shamser please review

File: ecl/hql/hqlmeta.cpp [hunk 1]
```diff
@@ -1838,16 +1843,16 @@ static IHqlExpression * createPreserveTableInfo(IHqlExpression * newTable, IHqlE
...
     IHqlExpression * groupSort = meta.groupSortOrder;
-    if (persistName && isKnownDistribution(distribution))
+    if (clusterSizeMayChange && isKnownDistribution(distribution))
     {
-        if (!distribution->isAttribute())
+        if (!distribution->isAttribute() || (distribution->queryName() == resizeAtom))
         {
             //Cluster size may not match so generate a unique modifier.  Needs to modify enough distribute no longer a nop,
             //but not too much to not get hoisted, or introduce extra dependencies.
             //At the moment bxor with a sequence number since I can't see anyone ever doing that.
```
Comment (richardkchapman) [inline ecl/hql/hqlmeta.cpp]: This comment is no longer true I think?

**Comment by richardkchapman** [comment]
@ghalliday One comment

File: ecl/hql/hqlmeta.cpp (hunk 1 above)
Comment (ghalliday) [inline ecl/hql/hqlmeta.cpp]: I wonder if this is the latest branch - I definitely modified that

**Comment by shamser** [comment]
@ghalliday Looks good.

**Comment by ghalliday** [comment]
@richardkchapman updated the comment

**Pull request merged**